src/pages/index.js

```diff
@@ -49,59 +32,168 @@ const popupCardForm = document.forms.newCardForm;
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    '.template-card',
+    handlePicturePopup,
+    () => {
+      popupDelete.setAction(() => {
+        api
+          .deleteCard(cardData._id)
+          .then(() => {
+            cardElement.remove();
+            popupDelete.close();
+          })
+          .catch((error) => {
+            console.error('Error deleting card:', error);
+          });
+      });
+      popupDelete.open();
+    },
+    handleLikeButton
+  );
+  const cardElement = card.getCardElement();
+  cardList.addItem(cardElement);
+}
+
+const popupDelete = new PopupConfirmation(popupConfirmDelete);
+
+/////
+function handleLikeButton(cardId) {
+  if (this.isLiked()) {
+    api
+      .dislikeCard(cardId)
+      .then((res) => this.setIsLiked(res.isLiked))
+      .catch((error) => {
+        console.error(error);
+      });
+  } else {
+    api
+      .likeCard(cardId)
+      .then((res) => this.setIsLiked(res.isLiked))
+      .catch((error) => {
+        console.error(error);
+      });
+  }
+}
+
+const loadingButtonText = 'Saving...';
+
+const popupNewCard = new PopupWithForm(newCard, handleNewCardSubmition, loadingButtonText);
+
+function handleNewCardSubmition(cardData) {
+  popupNewCard.renderLoading(true);
+  api
+    .addCard(cardData)
+    .then(() => {
+      createCard(cardData);
+    })
+    .finally(() => {
+      popupNewCard.renderLoading(false);
+      popupNewCard.reset();
+      popupNewCard.close();
+      newCardFormValidator.disableSubmitButton();
+    })
+    .catch((error) => {
+      console.error('Error adding a new card:', error);
+    });
+}
 
 btnNewCard.addEventListener('click', () => {
   popupNewCard.open();
 });
 
-const userInfo = new UserInfo({ nameSelector: '.profile__name', aboutMeSelector: '.profile__title' });
-
-const popupEditProfile = new PopupWithForm(editProfile, (data) => {
-  userInfo.setUserInfo(data);
+const userInfo = new UserInfo({ nameSelector: '.profile__name', aboutSelector: '.profile__title', avatarSelector: '.profile-photo-edit__image' });
+
+api
+  .getData()
+  .then(([userData, userCards]) => {
+    console.log(userData, userCards);
+    userInfo.setUserInfo(userData);
+    userInfo.setUserAvatar(userData);
+    cardList = new Section(
+      {
+        data: userCards,
+        renderer: createCard,
+      },
+      containerSelector // .cards
+    );
+    cardList.renderItems();
+  })
+  .catch((error) => {
+    console.error(error);
+  });
+
+const editProfileImage = new PopupWithForm(popupEditProfileImage, handleProfileImageSubmit, loadingButtonText);
+
+btnEditProfileImage.addEventListener('click', () => {
+  editProfileImage.open();
 });
 
+function handleProfileImageSubmit(userData) {
+  editProfileImage.renderLoading(true);
+  api
+    .updateProfileImage(userData)
+    .then(() => {
+      userInfo.setUserAvatar(userData);
+    })
+    .finally(() => {
+      editProfileImage.renderLoading(false);
+      editProfileImage.reset();
```

Comment:
Needs correction: you should reset the form, close the popup in `then` block, otherwise the form will close  even if an error occurred and you will need to reopen the form and fill the values again.